packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -427,6 +432,20 @@ async function loadCacheEntry(
 
   const currentTime = performance.timeOrigin + performance.now()
   if (
+    workUnitStore !== undefined &&
+    workUnitStore.type === 'prerender' &&
+    entry !== undefined &&
```

Comment:
One thing to note here is that we can't mark this if we don't have an existing cache entry. That's because for a miss we start streaming immediately.

In practice, because we drop in memory entries from the default Cache Handler using the `revalidate` time. We'll actually never hit this in the default cache.

We really need to handle this in the cacheScope but that's currently disabled pending another refactor to be able to store cache entries in the cacheScope.